### ISSUE TITLE
[Windows] It's not possible to update running server

### ISSUE THREAD

**Issue by rchl** [sublimelsp/LSP-vue#25]
On Windows files are locked when in use so on updating to new version, when trying to delete old `node_modules` directory an exception is thrown and server can be broken until restart.

I think we would need to be able to explicitly shut down server to fix that.

**Comment by rchl** [comment]
@rwols are you able to tell from the top of your head if this should work with the latest LSP?

**Comment by rwols** [comment]
I don't think this works.

**Comment by rchl** [comment]
I think it might be working because disabling a LSP plugin (which calls `plugin_unloaded() -> unregister_plugin()`) kills the server. So the same thing should happen during upgrading.

**Comment by rchl** [comment]
But then I should call `unregister_plugin()` before doing the cleanup in my refactor of lsp_utils.

**Comment by rwols** [comment]
> EDIT: But shutting down the server will happen async so after the cleanup anyway...

Yes. Which is why we need promise/futures/coroutines/awaitables. Slighty related to https://github.com/sublimehq/sublime_text/issues/3686
